lib/pure/ospaths.nim

```diff
@@ -532,6 +532,11 @@ when declared(getEnv) or defined(nimscript):
     tags: [ReadEnvEffect, ReadIOEffect].} =
     ## Returns the temporary directory of the current user for applications to
     ## save temporary files in.
+    when defined(tempDir):
+      const tempDir {.strdefine.}: string = nil
+      if not tempDir.isNil() and tempDir.len > 0:
```

Comment:
What if someone does `--define:useShPath:` (i.e. sets the sh path to the empty string)? Their own fault? I would be happy about that :P